luaui/Widgets/gui_gridmenu.lua

```diff
@@ -810,6 +814,30 @@ function reloadBindings()
 	else
 		Spring.SendCommands("bind " .. string.lower(PREV_PAGE_KEY) .. " gridmenu_prev_page")
 	end
+
+	local custom = {}
+	local useCustom = false
+	for r=1,3 do
+		local row = {}
+		for c=1,4 do
+			hotkey = 'gridmenu_' .. r .. '_' .. c
+			local key = Spring.GetActionHotKeys(hotkey)[1]
+			if key then
+				actionHotkey = Cfgs.keySymCharsReverse[string.upper(key)] or string.upper(key)
+				row[c] = actionHotkey
+				useCustom = true
+			else
+				row[c] = Cfgs.keyLayouts[currentLayout][r][c]
+			end
+		end
+		custom[4 - r] = row
+	end
+	Cfgs.keyLayouts['custom'] = custom
+	genKeyLayout('custom')
```

Comment:
should these be moved into `if useCustom`?

Reply:
Yes